Northstar.Coop/mod/scripts/vscripts/_respawns.nut

```diff
@@ -0,0 +1,339 @@
+}
+
+void function AddMakeSpecifcRespawns( string map, void functionref( entity ) func )
+{
+	file.CustomMapRespawns[map] <- func
+}
+
+
+void function StartSpawn( entity player )
+{
+    // Player was already positioned at info_player_start in SpPlayerConnecting.
+	// Don't reposition him, in case movers have already pushed him.
+	// No, I will
+
+	// log their UID
+	print( format( "%s : %s", player.GetPlayerName(), player.GetUID() ) )
+	
+	CheckPointInfo info = GetCheckPointInfo()
+
+	Chat_ServerPrivateMessage( player, "use 'say smth' in the console to chat ", false )
+
+	// for events
+	thread RunMiddleFunc( player )
+
+	if ( GetMapName() in file.CustomMapRespawns )
+	{
+		thread file.CustomMapRespawns[GetMapName()]( player )
+		return
+	}
+
+	else if ( info.pos != <0,0,0> )
+	{
+		player.SetOrigin( info.pos )
+
+		try
+		{
+			foreach ( int index, entity p in GetPlayerArray() )
+			{
+				if ( p == player )
+				{
+					compilestring( format("Map_PlayerDidLoad( GetPlayerArray()[%d] )", index ) ) // TODO: should be switched to OnClient_Connect where its uued
+				}
+			}
+		}
+		catch( aaa )
+		{
+			print( aaa )
+		}
+		
+		if ( !info.RsPilot )
+			thread MakePlayerTitan( player, info.pos )
+	}
+
+	DoRespawnPlayer( player, null )
+	// do we need this?
+	// AddPlayerMovementEventCallback( player, ePlayerMovementEvents.BEGIN_WALLRUN, Callback_WallrunBegin )
+	
+	// give them make specific items
+	OnTimeShiftGiveGlove( player )
+	if ( ( GetMapName() == "sp_beacon" || GetMapName() == "sp_beacon_spoke0" ) && Flag( "HasChargeTool" ) )
+		GiveBatteryChargeToolSingle( player )
+
+	// massive trol
+	for( int x = 1; x < achievements.MAX_ACHIVEMENTS; x++ )
+		UnlockAchievement( player, x )
+}
+
+void function RespawnPlayer( entity player )
+{
+	// really dangerous to enable this
+	// since everything is scripted
+	// stuff exepects people to be alive
+	// we to respawn them fast 
+	// WaitSignal( player, "RespawnNow" )
+
+	wait( 1 )
+
+	if ( !IsAlive( player ) && IsValid( player ) )
+    {
+		UpdateSpDifficulty( player )
+
+        printl("Respawning player:" + player.GetPlayerName() )
+
+		if ( GetPlayerArray().len() == 1 && file.RestartMap )
+            thread RestartMapWithDelay()
+        else if ( GetMapName() in file.CustomMapRespawns )
+            thread file.CustomMapRespawns[GetMapName()]( player )
+        else
+            thread GenericRespawn( player )
+	}
+}
+
+void function RestartMapWithDelay()
+{
+    wait 1
+    Coop_ReloadCurrentMapFromStartPoint( GetCurrentStartPointIndex() )
+}
+
+void function BasicRespawnLogic( entity player )
+{
+	player.kv.CollisionGroup = TRACE_COLLISION_GROUP_BLOCK_WEAPONS // remove collision between players ( I don't think it works XD )
+    player.s.inPostDeath = false
+    player.s.respawnSelectionDone = true
+    DoRespawnPlayer( player, null )
+    ClientCommand( player,"script_client thread MainHud_TurnOn_RUI( true )" ) // TODO: can this help with no hud on dedi?
```

Comment:
Didn't clientcommands get yeeted?

Reply:
Can't test rn, but this should no longer work on latest

Reply:
this might be the reason for the broken ui on  dedi servers :(